vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/parser/yaml/YamlVssParser.kt

```diff
@@ -62,64 +75,57 @@ internal class YamlVssParser(private val elementDelimiter: String = "") : VssPar
             .joinToString(separator = delimiter.toString())
             .substringAfter(delimiter) // Remove vssPath (already parsed)
             .prependIndent(delimiter.toString()) // So the parsing is consistent for the first element
-        val members = VssNodeSpecModel::class.memberProperties
-        val fieldsToSet = mutableListOf<Pair<String, Any?>>()
 
         // The VSSPath is an exception because it is parsed from the top level name.
-        val vssPathFieldInfo = Pair("vssPath", elementVssPath)
-        fieldsToSet.add(vssPathFieldInfo)
 
         // Parse (example: "description: Antilock Braking System signals.") into name + value for all .yaml lines
-        for (member in members) {
-            val memberName = member.name
-            if (!yamlElementJoined.contains(memberName)) continue
-
-            // Also parse the delimiter to not confuse type != datatype
-            val memberValue = yamlElementJoined
-                .substringAfter("$delimiter$memberName: ")
-                .substringBefore(delimiter)
-
-            val fieldInfo = Pair(memberName, memberValue)
-            fieldsToSet.add(fieldInfo)
-        }
-
-        val vssNodeSpec = VssNodeSpecModel()
-        vssNodeSpec.setFields(fieldsToSet)
-
-        if (vssNodeSpec.uuid.isEmpty()) return null
-
-        return vssNodeSpec
+        val uuid = fetchValue(KEY_DATA_UUID, yamlElementJoined, delimiter)
+            ?: throw FileParseException("Could not parse '$KEY_DATA_UUID' for '$vssPath'")
+
+        val type = fetchValue(KEY_DATA_TYPE, yamlElementJoined, delimiter)
+            ?: throw FileParseException("Could not parse '$KEY_DATA_TYPE' for '$vssPath'")
+
+        val description = fetchValue(KEY_DATA_DESCRIPTION, yamlElementJoined, delimiter) ?: ""
+        val datatype = fetchValue(KEY_DATA_DATATYPE, yamlElementJoined, delimiter) ?: ""
+        val comment = fetchValue(KEY_DATA_COMMENT, yamlElementJoined, delimiter) ?: ""
+        val unit = fetchValue(KEY_DATA_UNIT, yamlElementJoined, delimiter) ?: ""
+        val min = fetchValue(KEY_DATA_MIN, yamlElementJoined, delimiter) ?: ""
+        val max = fetchValue(KEY_DATA_MAX, yamlElementJoined, delimiter) ?: ""
+
+        val vssDataType = VssDataType.find(datatype)
+        val valueDataType = vssDataType.valueDataType
+
+        val vssNodeProperties = mutableSetOf(
+            VssNodeProperty(vssPath, KEY_DATA_UUID, uuid, String::class),
+            VssNodeProperty(vssPath, KEY_DATA_TYPE, type, String::class),
+            VssNodeProperty(vssPath, KEY_DATA_DESCRIPTION, description, String::class),
+            VssNodeProperty(vssPath, KEY_DATA_COMMENT, comment, String::class),
+            VssSignalProperty(vssPath, KEY_DATA_DATATYPE, datatype, valueDataType),
+            VssSignalProperty(vssPath, KEY_DATA_UNIT, unit, String::class),
+            VssSignalProperty(vssPath, KEY_DATA_MIN, min, valueDataType),
+            VssSignalProperty(vssPath, KEY_DATA_MAX, max, valueDataType),
```

Comment:
Could we maybe replace this somehow with a builder? Would it be easier and clearer to use?

VssNodePropertiesBuilder(uuid, type).withDescription(...).withComment(...).withDataType(...).withMin(value, type)... returning a Set<VssNodeProperty>